source/APILookupGdk.txt

```diff
@@ -61,3 +61,51 @@ code: end
 struct: Surface
 import: cairo.Surface : CairoSurface = Surface
 structWrap: cairo.Surface CairoSurface
+
+struct: Texture
+noCode: new_from_resource
+noCode: new_from_filename
+code: start
+	/**
+	 * Creates a new texture by loading an image from a file or a resource.
+	 *
+	 * The file format is detected automatically. The supported formats
+	 * are PNG and JPEG, though more formats might be available.
+	 *
+	 * If %NULL is returned, then @error will be set.
+	 *
+	 * This function is threadsafe, so that you can e.g. use GTask
+	 * and g_task_run_in_thread() to avoid blocking the main thread
+	 * while loading a big image.
+	 *
+	 * Params:
+	 *     path = the filename or resourcename to load
+	 *     resource = true if load from resource
+	 *
+	 * Returns: A newly-created `GdkTexture`
+	 *
+	 * Since: 4.6
+	 *
+	 * Throws: GException on failure.
+	 * Throws: ConstructionException GTK+ fails to create the object.
+	 */
+	public this(string path, bool resource = false)
+	{
+		GError* err = null; GdkTexture *__p;
```

Comment:
double underscores are reserved for language internal stuff, you might want to just use `_p` or a simple `p` here as this is local function (constructor) scope anyway.

Compiler generated helper variables could otherwise clash with this.